### ISSUE TITLE
Error Compiling Granne

### ISSUE THREAD

**Issue by os-gabe** [granne/granne#12]
I am attempting to move over to the new `v0.5.0` release but am encountering this error while trying to get it setup:
```
$ pip install -e git://github.com/granne/granne.git@v0.5.0#egg=granne
Obtaining granne from git+git://github.com/granne/granne.git@v0.5.0#egg=granne
  Updating /home/gabe/Envs/granne-test/src/granne clone (to revision v0.5.0)
  Running command git fetch -q --tags
  Running command git reset --hard -q 662656d2cb0aa43558d826e7f1e7568afb2ea36d
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
Installing collected packages: granne
  Running setup.py develop for granne
    ERROR: Command errored out with exit status 1:
     command: /home/gabe/Envs/granne-test/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/gabe/Envs/granne-test/src/granne/setup.py'"'"'; __file__='"'"'/home/gabe/Envs/granne-test/src/granne/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /home/gabe/Envs/granne-test/src/granne/
    Complete output (123 lines):
    running develop
    running egg_info
    writing granne.egg-info/PKG-INFO
    writing dependency_links to granne.egg-info/dependency_links.txt
    writing top-level names to granne.egg-info/top_level.txt
    writing manifest file 'granne.egg-info/SOURCES.txt'
    running build_ext
    running build_rust
           Fresh autocfg v1.0.0
           Fresh cfg-if v0.1.10
           Fresh lazy_static v1.4.0
           Fresh scopeguard v1.1.0
           Fresh regex-syntax v0.6.17
           Fresh smallvec v1.2.0
           Fresh ppv-lite86 v0.2.6
           Fresh adler32 v1.0.4
           Fresh itoa v0.4.5
           Fresh stable_deref_trait v1.1.1
           Fresh either v1.5.3
           Fresh thread_local v1.0.1
           Fresh lock_api v0.3.3
       Compiling granne v0.5.0 (/home/gabe/Envs/granne-test/src/granne)
         Running `/home/gabe/Envs/granne-test/src/granne/target/debug/build/granne-3065d63cc7adb6bb/build-script-build`
           Fresh libc v0.2.68
           Fresh maybe-uninit v2.0.0
           Fresh byteorder v1.3.4
           Fresh ryu v1.0.3
           Fresh serde v1.0.105
           Fresh memchr v2.3.3
           Fresh crc32fast v1.2.0
           Fresh num-traits v0.2.11
           Fresh getrandom v0.1.14
           Fresh num_cpus v1.12.0
           Fresh time v0.1.42
           Fresh parking_lot_core v0.7.0
           Fresh memmap v0.7.0
           Fresh madvise v0.1.0
           Fresh miniz_oxide v0.3.6
           Fresh owning_ref v0.4.1
           Fresh crossbeam-utils v0.7.2
           Fresh memoffset v0.5.4
           Fresh crossbeam-queue v0.2.1
           Fresh rand_core v0.5.1
           Fresh rand_chacha v0.2.2
           Fresh ordered-float v1.0.2
           Fresh aho-corasick v0.7.10
           Fresh crossbeam-epoch v0.8.2
           Fresh pbr v1.0.2
           Fresh parking_lot v0.10.0
           Fresh flate2 v1.0.14
           Fresh stream-vbyte v0.3.2
           Fresh serde_json v1.0.48
           Fresh fxhash v0.2.1
           Fresh regex v1.3.5
           Fresh crossbeam-deque v0.7.3
           Fresh rand v0.7.3
           Fresh rayon-core v1.7.0
           Fresh rayon v1.3.0
           Fresh python3-sys v0.4.1
           Fresh cpython v0.4.1
         Running `rustc --crate-name granne --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 --cfg 'feature="default"' -C metadata=7d4dec9265969171 -C extra-filename=-7d4dec9265969171 --out-dir /home/gabe/Envs/granne-test/src/granne/target/debug/deps -C incremental=/home/gabe/Envs/granne-test/src/granne/target/debug/incremental -L dependency=/home/gabe/Envs/granne-test/src/granne/target/debug/deps --extern byteorder=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libbyteorder-0a48faa69a27c0a8.rmeta --extern flate2=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libflate2-5c9df00987ebb4d3.rmeta --extern fxhash=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libfxhash-a9584c0803658a1c.rmeta --extern madvise=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libmadvise-2bff7c02f1d9a0f6.rmeta --extern memmap=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libmemmap-3467cfb5935cea6b.rmeta --extern ordered_float=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libordered_float-49b58c9dad452031.rmeta --extern owning_ref=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libowning_ref-88449d183d98535d.rmeta --extern parking_lot=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libparking_lot-e29d0e72287e3d37.rmeta --extern pbr=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libpbr-df0b03d13e7090cb.rmeta --extern rand=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/librand-34f070eb1e7a7deb.rmeta --extern rayon=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/librayon-da71c033c21c80ff.rmeta --extern serde_json=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libserde_json-5fd5b5f28b6a28d5.rmeta --extern stream_vbyte=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libstream_vbyte-92918a05764be3c1.rmeta`
    error[E0308]: mismatched types
      --> src/elements/dense_vector.rs:86:17
       |
    86 |                 Self(self.0.into_owned())
       |                 ^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
       |
      ::: src/elements/angular.rs:53:1
       |
    53 | dense_vector!(f32);
       | ------------------- in this macro invocation
       |
       = note: expected struct `elements::angular::Vectors<'static>`
                  found struct `elements::angular::Vectors<'a>`
    note: the lifetime `'a` as defined on the impl at 40:14...
      --> src/elements/dense_vector.rs:40:14
       |
    40 |         impl<'a> Vectors<'a> {
       |              ^^
       |
      ::: src/elements/angular.rs:53:1
       |
    53 | dense_vector!(f32);
       | ------------------- in this macro invocation
       = note: ...does not necessarily outlive the static lifetime
       = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
    
    error[E0308]: mismatched types
      --> src/elements/dense_vector.rs:86:17
       |
    86 |                 Self(self.0.into_owned())
       |                 ^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
       |
      ::: src/elements/angular_int.rs:17:1
       |
    17 | dense_vector!(i8);
       | ------------------ in this macro invocation
       |
       = note: expected struct `elements::angular_int::Vectors<'static>`
                  found struct `elements::angular_int::Vectors<'a>`
    note: the lifetime `'a` as defined on the impl at 40:14...
      --> src/elements/dense_vector.rs:40:14
       |
    40 |         impl<'a> Vectors<'a> {
       |              ^^
       |
      ::: src/elements/angular_int.rs:17:1
       |
    17 | dense_vector!(i8);
       | ------------------ in this macro invocation
       = note: ...does not necessarily outlive the static lifetime
       = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
    
    error: aborting due to 2 previous errors
    
    For more information about this error, try `rustc --explain E0308`.
    error: could not compile `granne`.
    
    Caused by:
      process didn't exit successfully: `rustc --crate-name granne --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 --cfg 'feature="default"' -C metadata=7d4dec9265969171 -C extra-filename=-7d4dec9265969171 --out-dir /home/gabe/Envs/granne-test/src/granne/target/debug/deps -C incremental=/home/gabe/Envs/granne-test/src/granne/target/debug/incremental -L dependency=/home/gabe/Envs/granne-test/src/granne/target/debug/deps --extern byteorder=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libbyteorder-0a48faa69a27c0a8.rmeta --extern flate2=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libflate2-5c9df00987ebb4d3.rmeta --extern fxhash=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libfxhash-a9584c0803658a1c.rmeta --extern madvise=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libmadvise-2bff7c02f1d9a0f6.rmeta --extern memmap=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libmemmap-3467cfb5935cea6b.rmeta --extern ordered_float=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libordered_float-49b58c9dad452031.rmeta --extern owning_ref=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libowning_ref-88449d183d98535d.rmeta --extern parking_lot=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libparking_lot-e29d0e72287e3d37.rmeta --extern pbr=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libpbr-df0b03d13e7090cb.rmeta --extern rand=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/librand-34f070eb1e7a7deb.rmeta --extern rayon=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/librayon-da71c033c21c80ff.rmeta --extern serde_json=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libserde_json-5fd5b5f28b6a28d5.rmeta --extern stream_vbyte=/home/gabe/Envs/granne-test/src/granne/target/debug/deps/libstream_vbyte-92918a05764be3c1.rmeta` (exit code: 1)
    cargo rustc --lib --manifest-path py/Cargo.toml --features cpython/extension-module cpython/python3-sys --verbose -- --crate-type cdylib
    error: cargo failed with code: 101
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/gabe/Envs/granne-test/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/gabe/Envs/granne-test/src/granne/setup.py'"'"'; __file__='"'"'/home/gabe/Envs/granne-test/src/granne/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

```
I don't know rust so I'm a little lost with how to debug this.

**Comment by os-gabe** [comment]
Interestingly
`cargo build --release`
and
`cargo test`
complete successfully

**Comment by os-gabe** [comment]
Oops, looks like you already fixed this in #11

**Comment by erik-cliqz** [comment]
I created a new release for this now including the fix: [v0.5.1](https://github.com/granne/granne/releases/tag/v0.5.1).
